pcdsdevices/gon.py

```diff
@@ -525,6 +525,16 @@ def forward(self, pseudo_pos):
                                       pseudo_pos.e_phi)
         return self.RealPosition(eta=eta, kappa=kappa, phi=phi)
 
+    @pseudo_position_argument
+    def move(self, position, wait=True, timeout=None, moved_cb=None):
+        if self.check_motor_step(position.e_eta, position.e_chi,
+                                 position.e_phi):
+            return super().move(position, wait=wait, timeout=timeout,
+                                moved_cb=moved_cb)
+        else:
+            logger.warning('Aborting moving for safety.')
+            return
```

Comment:
This should return a failed status object, maybe like:
```suggestion
            status = DeviceStatus(self)
            status.set_exception(ValueError('Unsafe Kappa move aborted!'))
            return status
```

Reply:
I did not test this snippit! So I'm sorry if it doesn't quite work right

Reply:
The API of the `move` method is to return a status object of some kind, I should have mentioned that first in this comment chain